spec/services/authentication_service_spec.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe AuthenticationService do
+  describe '#process' do
+    let(:mobile_number) { '9288402094' }
+    let(:otp) { '299493' }
+
+    context 'when there is no matching record in OtpRequest for the given mobile number and otp combination' do
+      it 'returns an error' do
+        service = described_class.new(mobile_number, otp)
+        response = service.process
+
+        expect(response.token).to be_nil
+        expect(response.error).to eq('Could not find a matching OTP code')
+      end
+    end
+
+    context 'when there exists an OtpRequest record for the given mobile number and otp but it is expired' do
+      before do
+        create(:otp_request, mobile_number: mobile_number, otp: otp, expire_at: 5.hours.ago)
+      end
+
+      it 'returns an error' do
+        service = described_class.new(mobile_number, otp)
+        response = service.process
+
+        expect(response.token).to be_nil
+        expect(response.error).to eq('OTP is expired')
+      end
+    end
+
+    context 'when the OTP verification is successful' do
+      before do
+        create(:otp_request, mobile_number: mobile_number, otp: otp, expire_at: 10.minutes.from_now)
+      end
+
+      context 'with no user record present in db having the given mobile number' do
+        it 'creates a user and return the JWT token generated' do
+          service = described_class.new(mobile_number, otp)
```

Comment:
Optional:

 Can we use `[subject](https://github.com/rubocop/rspec-style-guide/issues/6)` instead of creating a variable. This is a common RSpec pattern and makes the test cleaner.

Reply:
This spec is outdated now. 

I still have a spec for expired? method. But I cannot change it to use subject as my instance is defined under the `it` block. (I can't move it to the describe / context blocks as the attributes are different for different `it` blocks)